### PR TITLE
Remove model.predict

### LINKED ISSUE #1004: [RFC] Deprecate `model.predict` in favour of `Trainer.predict`
`model.predict` has a lot of usability issues:
- not clear how to load from different inputs
- no way to pass additional data loading arguments
- no docstring specific to the particular task
- lots of magic in the background to resolve e.g. transforms and stuff like that
- not able to handle much load (see #983)
- two ways to do prediction

We should deprecate in favour of using a datamodule and Trainer. E.g. image classification prediction is currently:
```
predictions = model.predict(
    [
        "data/hymenoptera_data/val/bees/65038344_52a45d090d.jpg",
        "data/hymenoptera_data/val/bees/590318879_68cf112861.jpg",
        "data/hymenoptera_data/val/ants/540543309_ddbb193ee5.jpg",
    ]
)
```

Would become:
```
predict_data_module = ImageClassificationData.from_files(predict_files=[
    "data/hymenoptera_data/val/bees/65038344_52a45d090d.jpg",
    "data/hymenoptera_data/val/bees/590318879_68cf112861.jpg",
    "data/hymenoptera_data/val/ants/540543309_ddbb193ee5.jpg",
])
predictions = trainer.predict(model, datamodule=predict_data_module)
```

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes #1004 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/model.py [hunk 1]
```diff
@@ -472,63 +470,6 @@ def test_step(self, batch: Any, batch_idx: int) -> None:
             prog_bar=True,
         )
 
-    @predict_context
-    def predict(
```
Comment (tchaton) [inline flash/core/model.py:476]: Love it !

File: flash/image/face_detection/data.py [hunk 2]
```diff
@@ -174,3 +169,29 @@ def from_datasets(
...
+    @classmethod
+    def from_files(
+        cls,
+        predict_files: Optional[Sequence[str]] = None,
+        predict_transform: Optional[Dict[str, Callable]] = None,
+        **data_module_kwargs: Any,
+    ) -> "FaceDetectionData":
+        return cls(
+            predict_dataset=ImageClassificationFilesInput(RunningStage.PREDICTING, predict_files),
+            input_transform=cls.input_transform_cls(predict_transform=predict_transform),
```
Comment (tchaton) [inline flash/image/face_detection/data.py:182]: We will have to move those to the new API in a future PR. Just a reminder.

File: flash_examples/integrations/labelstudio/video_classification.py [hunk 3]
```diff
@@ -29,8 +29,8 @@
 trainer.finetune(model, datamodule=datamodule, strategy="freeze")
 
 # 5. Make a prediction
-predictions = model.predict(os.path.join(os.getcwd(), "data/test"))
-print(predictions)
+predict_datamodule = VideoClassificationData.from_folders(predict_folder=os.path.join(os.getcwd(), "data/test"))
```
Comment (tchaton) [inline flash_examples/integrations/labelstudio/video_classification.py:32]: I prefer to keep datamodule everything.

**Review by tchaton — APPROVED** [review]
LGTM !

File: flash_examples/integrations/labelstudio/video_classification.py (hunk 3 above)
Comment (ethanwharris) [reply flash_examples/integrations/labelstudio/video_classification.py]: Done :smiley:

**Pull request merged**